test/ownership.js

```diff
@@ -4,154 +4,124 @@ const TUPELO_HOST = process.env.TUPELO_RPC_HOST || 'localhost:50051'
-        // TODO: clear RPC server state between tests automatically
-        // await bob.register();
-
-        let resp = await bob.generateKey();
-        const bobKey = resp.keyAddr;
-
-        resp = await alice.generateKey();
-        const aliceKey = resp.keyAddr;
-
-        let {chainId,} = await alice.createChainTree(aliceKey);
-
-        for (let i = 0; i < 5; i++) {
-            resp = await alice.setData(chainId, aliceKey, "path/to/" + i.toString(), "value: " + i.toString());
-            assert.notEqual(resp.tip, null);
-        }
-
-        // make sure all sets can be read back
-        for (let i = 0; i < 5; i++) {
-            let result = await alice.resolveData(chainId, "path/to/" + i.toString());
-            let oldResult = await alice.resolve(chainId, "path/to/" + i.toString());
-            assert.ok(result.data == "value: " + i.toString() || oldResult.data == "value: " + i.toString());
-        }
-
-        // transfer ownership to be shared with Alice and bob
-        resp = await alice.setOwner(chainId, aliceKey, [aliceKey, bobKey]);
-        assert.notEqual(resp.tip, null);
-
-        // send the chaintree over to bob
-        let chainTreeExport = await alice.exportChainTree(chainId);
-
-        resp = await bob.importChainTree(chainTreeExport.chainTree);
-        assert.equal(resp.chainId, chainId)
-
-        // make sure bob can read all the previous history
-        for (let i = 0; i < 5; i++) {
-            let result = await bob.resolveData(chainId, "path/to/" + i.toString());
-            let oldResult = await bob.resolve(chainId, "path/to/" + i.toString());
-            assert.ok(result.data == "value: " + i.toString() || oldResult.data == "value: " + i.toString());
-        }
-
-        // and can himself write to the tree
-        resp = await bob.setData(chainId, bobKey, "path/to/bobvalue", "bobdidthis");
-        assert.notEqual(resp.tip, null);
-
-        result = await bob.resolveData(chainId, "path/to/bobvalue");
-        oldResult = await bob.resolve(chainId, "path/to/bobvalue");
-        assert.ok(result.data == "bobdidthis" || oldResult.data == "bobdidthis");
-
-        return Promise.resolve(true);
+    // TODO: clear RPC server state between tests automatically
+    await bob.register();
+    const {keyAddr: bobKey,} = await bob.generateKey();
+    assert.strictEqual(bobKey.length, 42);
+    
+    let resp = await alice.setData(chainId, aliceKey, "path/to/here", "hi");
+    assert.notEqual(resp.tip, null);
+    
+    let result = await alice.resolveData(chainId, "path/to/here");
+    assert.deepStrictEqual(result.data, ["hi",]);
+    
+    const chainTreeExport = await alice.exportChainTree(chainId);
+    resp = await bob.importChainTree(chainTreeExport.chainTree);
+    assert.strictEqual(resp.chainId, chainId)
+    
+    result = await bob.resolveData(chainId, "path/to/here");
+    assert.deepStrictEqual(result.data, ["hi",]);
+  });
+  
+  it("can transfer from alice to bob", async () => {
```

Comment:
is this test useful anymore?

Reply:
@tobowers Good question, but I can't tell, I don't know its context.

Reply:
you're looking at its context as much as I am (and I wrote it). I believe we added the "real" bit because ownership was working but the data wasn't coming along for a ride. Now that we have more robust tests this one seems redundant to me.

Reply:
@tobowers I'm not sure if its useful when considered in relation to the "can do a real transfer from alice to bob" test case, if that's what you're thinking of.

Reply:
@tobowers I can drop that test from this PR then.

Reply:
Pushed an update with this test removed.